config/Dockerfiles/linchpin-tests.sh

```diff
@@ -1,31 +1,158 @@
-#!/bin/bash -x
+#!/bin/bash
 
-base_dir="$( pwd )"
+base_dir="${PWD}"
 
 set -o pipefail
 
 mkdir -p ${base_dir}/${distro}_logs
+rm -f ${base_dir}/${distro}_logs/*
 
-result=0
-for i in $TARGETS; do
-    testname="$distro/$i"
-    export CREDS_PATH="$base_dir/keys/$i"
-    # If CREDS_PATH provides a tarball extract it and
-    # run it's install script
-    if [ -e "$CREDS_PATH/${i}.tgz" ]; then
-        tmpdir=$(mktemp -d)
-        tar xvf $CREDS_PATH/${i}.tgz -C $tmpdir
-        $tmpdir/install.sh
+TESTS_DIR="./config/Dockerfiles/tests.d"
+
+
+if [ -z "${PROVIDERS}" ]; then
+    PROVIDERS="dummy libvirt"
+fi
+
+# this function checks the distros.exclude value within the test.
+# If none, false is returned and the disro hasn't been excluded.
+# If distros.exclude includes a distro name, it is chedked against
+# the ${distro} env var. If they match true is sent back to the caller
+# and the test will be skipped for that distro.
+function check_distro_exclude () {
+    shopt -s extglob
+    test=${1}
+    dist_exc=$(grep 'distros.exclude' ${test} | grep -v 'none' | awk -F':' '{ print $2 }')
```

Comment:
I like the way this works.  I think it should be more strict though?  Maybe '^# distros.exclude:' ?  Same for the inlcudes below...

Reply:
Ok, that sounds good. I'll update the regexp on that one. 

On the TMP_PF below, I've updated those in my latest commits, just haven't pushed them yet, so you should see that change shortly.